src/test_l3_dsDisplay.c

```diff
@@ -0,0 +1,435 @@
+
+    UT_LOG_INFO("Result dsGetDisplay(Handle:[0x%0X]]) dsError_t=[%s]", displayHandle, UT_Control_GetMapString(errorMappingTable, status));
+    DS_ASSERT(status == dsERR_NONE);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+
+}
+
+/**
+ * @brief Retrieves Extended Display Identification Data (EDID) for the display.
+ *
+ * Retrieves the EDID information for the display associated with the selected handle.
+ *
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 004@n
+ *
+ * **Test Procedure:**
+ * Refer to Test specification documentation
+ * [ds-display_L3_Low-Level_TestSpecification.md](../../docs/pages/ds-display_L3_Low-Level_TestSpecification.md)
+ */
+void dsDisplay_GetEDID() 
+{
+    gTestID = 4;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    dsDisplayEDID_t *edid =NULL;
+
+    dsGetDisplay_getHandle();
+
+    if (displayHandle == -1) {
+        UT_LOG_ERROR("Failed to get display handle\n");
+        return;
+    }
+
+    edid = (dsDisplayEDID_t *)malloc(sizeof(dsDisplayEDID_t));
+    if (edid == NULL) {
+        UT_LOG_ERROR("Failed to allocate memory for edid\n");
+        return;
+    }
+
+    UT_LOG_INFO("Calling dsGetEDID(IN:handle:[0x%0X]])", displayHandle);
+    status = dsGetEDID(displayHandle, edid);
+
+    if (status != dsERR_NONE) {
+        UT_LOG_ERROR("dsGetEDID failed with error: %d", status);
+        free(edid);
+        return;
+    }
+
+    UT_LOG_INFO("Result dsGetEDID(dsDisplayEDID_t(productCode:[%d], serialNumber:[%d], manufactureYear:[%d], manufactureWeek:[%d], hdmiDeviceType:[%s], isRepeater:[%s], physicalAddressA:[%u], physicalAddressB:[%u], physicalAddressC:[%u], physicalAddressD:[%u], numOfSupportedResolution:[%d], monitorName:[%s])", \
+             edid->productCode, edid->serialNumber, edid->manufactureYear, edid->manufactureWeek, edid->hdmiDeviceType, edid->isRepeater, edid->physicalAddressA, edid->physicalAddressB, edid->physicalAddressC, edid->physicalAddressD, edid->numOfSupportedResolution, edid->monitorName);
+    DS_ASSERT(status == dsERR_NONE);
+
+    free(edid);
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+ * @brief Retrieves the Extended Display Identification Data (EDID) bytes for the display.
+ *
+ * Retrieves the EDID data bytes for the display associated with the selected handle.
+ *
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 005@n
+ *
+ * **Test Procedure:**
+ * Refer to Test specification documentation
+ * [ds-display_L3_Low-Level_TestSpecification.md](../../docs/pages/ds-display_L3_Low-Level_TestSpecification.md)
+ */
+void dsDisplay_GetEDIDBytes()
+{
+    gTestID = 5;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    int length = 0;
+    unsigned char *edid = NULL;
+
+    dsGetDisplay_getHandle();
+
+    UT_LOG_INFO("Calling dsGetEDID(Handle:[0x%0X]])", displayHandle);
+    status = dsGetEDIDBytes(displayHandle, edid, &length);
+
```

Comment:
print the edid bytes in result logging to communicate to user/python

Reply:
Done